compiler-core/src/strings.rs

```diff
@@ -64,3 +66,57 @@ pub fn convert_string_escape_chars(str: &EcoString) -> EcoString {
     }
     filtered_str
 }
+
+/// Converts a string into its UTF-16 representation in bytes
+pub fn string_to_utf16_bytes(string: &str, endianness: Endianness) -> Vec<u8> {
+    let mut bytes = Vec::with_capacity(string.len() * 2);
+
+    let mut character_buffer = [0, 0];
+    for character in string.chars() {
+        let segments = character.encode_utf16(&mut character_buffer);
+
+        for segment in segments {
+            let segment_bytes = match endianness {
+                Endianness::Big => segment.to_be_bytes(),
+                Endianness::Little => segment.to_le_bytes(),
+            };
+
+            bytes.push(segment_bytes[0]);
+            bytes.push(segment_bytes[1]);
+        }
+    }
+
+    bytes
+}
+
+/// Converts a string into its UTF-32 representation in bytes
+pub fn string_to_utf32_bytes(string: &str, endianness: Endianness) -> Vec<u8> {
+    let mut bytes = Vec::with_capacity(string.len() * 4);
+
+    for character in string.chars() {
+        let character_bytes = match endianness {
+            Endianness::Big => (character as u32).to_be_bytes(),
+            Endianness::Little => (character as u32).to_le_bytes(),
+        };
+        bytes.extend(character_bytes);
+    }
+
+    bytes
+}
+
+/// Gets the number of UTF-16 codepoints it would take to encode a given string.
+pub fn length_utf16(string: &str) -> usize {
+    let mut length = 0;
+
+    for char in string.chars() {
+        length += char.len_utf16()
+    }
+
+    length
+}
+
+/// Gets the number of UTF-32 codepoints in a string (also known as the number of
+/// characters).
```

Comment:
We deliberately avoid the word "characters" in Gleam because it's vague, but when we do use it we use it to mean graphemes as that's closer to how humans think.